From 008f990584171f7a00d022f7b995c47a6df84070 Mon Sep 17 00:00:00 2001
From: genius user <genius@ReliableEmbeddedSystems.com>
Date: Sat, 13 May 2017 06:49:04 +0000
Subject: [PATCH] phytec-mira: dts/Makefile

Signed-off-by: genius user <genius@ReliableEmbeddedSystems.com>
---
 arch/arm/boot/dts/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 7037201..a8beca2 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -358,6 +358,7 @@ dtb-$(CONFIG_SOC_IMX6Q) += \
 	imx6dl-tx6u-81xx-mb7.dtb \
 	imx6dl-udoo.dtb \
 	imx6dl-wandboard.dtb \
+	imx6q-phytec-mira-rdk-nand.dtb \
 	imx6dl-wandboard-revb1.dtb \
 	imx6q-apalis-ixora.dtb \
 	imx6q-apf6dev.dtb \
-- 
2.7.4

