From 5f5e1445073f1644033a9730490c82f7f4ff18af Mon Sep 17 00:00:00 2001
From: student-bbbio <student-bbbio@ReliableEmbeddedSystems.com>
Date: Wed, 3 Feb 2016 23:39:26 +0200
Subject: [PATCH] QSPI, uEnv.txt from SD card

Signed-off-by: student-bbbio <student-bbbio@ReliableEmbeddedSystems.com>
---
 include/configs/zynq-common.h | 29 ++++++++++++++++++++++++++++-
 1 file changed, 28 insertions(+), 1 deletion(-)

diff --git a/include/configs/zynq-common.h b/include/configs/zynq-common.h
index 0ab6083..211d4a3 100644
--- a/include/configs/zynq-common.h
+++ b/include/configs/zynq-common.h
@@ -199,6 +199,8 @@
 #ifndef CONFIG_ENV_IS_NOWHERE
 # ifndef CONFIG_SYS_NO_FLASH
 #  define CONFIG_ENV_IS_IN_FLASH
+# elif defined(CONFIG_ZYNQ_QSPI)
+#  define CONFIG_ENV_IS_IN_SPI_FLASH
 # elif defined(CONFIG_SYS_NO_FLASH)
 #  define CONFIG_ENV_IS_NOWHERE
 # endif
@@ -207,6 +209,9 @@
 # define CONFIG_ENV_OFFSET		0xE0000
 #endif
 
+/* RES custom stuff! */
+#define CONFIG_CMD_ZYNQ_UENVBOOT        /* if $modeboot SD card + uEnv.txt on SD card */
+
 /* Default environment */
 #define CONFIG_EXTRA_ENV_SETTINGS	\
 	"fit_image=fit.itb\0"		\
@@ -230,9 +235,31 @@
 			"load usb 0 ${load_addr} ${fit_image} && " \
 			"bootm ${load_addr}\0" \
 		"fi\0" \
+        "loadbootenv_addr=0x2000000\0" \
+        "bootenv=uEnv.txt\0" \
+	"loadbootenv=load mmc 0 ${loadbootenv_addr} ${bootenv}\0" \
+	"importbootenv=echo Importing environment from SD ...; " \
+		"env import -t ${loadbootenv_addr} $filesize\0" \
+ 	"sd_uEnvtxt_existence_test=test -e mmc 0 /uEnv.txt\0" \
+        "uenvboot=if test $modeboot = sdboot && env run sd_uEnvtxt_existence_test; " \
+			"then if env run loadbootenv; " \
+				"then env run importbootenv; " \
+			"fi; " \
+                        "if test -n $uenvcmd; then " \
+                                "echo Running uenvcmd ...; " \
+                                "run uenvcmd; " \
+                        "fi; "\
+		"fi; \0" \
 		DFU_ALT_INFO
 
-#define CONFIG_BOOTCOMMAND		"run $modeboot"
+
+/* default boot is according to the bootmode switch settings or SD card + uEnv.txt */
+#if defined(CONFIG_CMD_ZYNQ_UENVBOOT)
+#define CONFIG_BOOTCOMMAND              "run uenvboot"
+#else
+#define CONFIG_BOOTCOMMAND              "run $modeboot"
+#endif
+
 #define CONFIG_BOOTDELAY		3 /* -1 to Disable autoboot */
 #define CONFIG_SYS_LOAD_ADDR		0 /* default? */
 
-- 
1.9.1

