From 4c62e4620d5548bc00a1746317dbecbb9c91884a Mon Sep 17 00:00:00 2001
From: Robert Berger <robert.berger@ReliableEmbeddedSystems.com>
Date: Mon, 2 Oct 2017 15:58:17 +0000
Subject: [PATCH] mariadb: fix gcc7 compile

Signed-off-by: Robert Berger <robert.berger@ReliableEmbeddedSystems.com>
---
 client/mysql.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/client/mysql.cc b/client/mysql.cc
index 9d255b5..be436cb 100644
--- a/client/mysql.cc
+++ b/client/mysql.cc
@@ -2849,7 +2849,7 @@ You can turn off this feature to get a quicker startup with -A\n\n");
         mysql_free_result(fields);
         break;
       }
-      field_names[i][num_fields*2]= '\0';
+      field_names[i][num_fields*2]= NULL;
       j=0;
       while ((sql_field=mysql_fetch_field(fields)))
       {
-- 
2.7.4

